src/Pretzel/Commands/PretzelCommandHandler.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Linq;
+using System.CommandLine.Binding;
+using System.CommandLine.Invocation;
+using System.Composition;
+using System.Threading.Tasks;
+using Pretzel.Logic;
+using Pretzel.Logic.Commands;
+
+namespace Pretzel.Commands
+{
+    public class PretzelCommandHandler : ICommandHandler
+    {
+        public ICommandParameters CommandParameters { get; }
+        public ExportFactory<IPretzelCommand, CommandInfoAttribute> Command { get; }
+        public IConfiguration Configuration { get; }
+
+        public PretzelCommandHandler(IConfiguration configuration, ICommandParameters commandParameters, ExportFactory<IPretzelCommand, CommandInfoAttribute> command)
+        {
+            Configuration = configuration;
+            CommandParameters = commandParameters;
+            Command = command;
+        }
+
+        public async Task<int> InvokeAsync(InvocationContext context)
+        {
+            var bindingContext = context.BindingContext;
+
+            if (CommandParameters != null)
+            {
+                new ModelBinder(CommandParameters.GetType())
+                   .UpdateInstance(CommandParameters, bindingContext);
+
+                CommandParameters.BindingCompleted();
+            }
+
+            if (CommandParameters is ISourcePathProvider pathProvider)
+            {
+                Configuration.ReadFromFile(pathProvider.Source);
```

Comment:
If the command isn't an `ISourcePathProvider` the configuration is not loaded?
We should have a default source for that case.

Reply:
No if the argument that is used is no `ISourcePathProvider` then no configuration is read. That's only possible in 2 cases: The command does not use arguments at all. A plugin author decides to implement ICommandArgument by hand, ignoring the fact that a `PretzelBaseCommandArguments` class should be used.
The default's provided by `DefaultsConfiguration` are still there.

Reply:
If I am not mistaken, in these cases even if there is a configuration file, it is not loaded?
That is not a thing we want, it is very confusing.

Reply:
It's not loaded in 2 cases: For the `VersionCommand` and the `HungryCommand`. 
Both provide no extension support. All other Commands provide this. 
If I think further a bit there are commands where the configuration don't need to be read, for example the nuget ones. I think this is okay for plugin authors to use `PretzelBaseCommandArguments` for their custom commands if they want to extend the site, and go with `BaseCommandArguments` (or implement the interfaces them self) if they want to get lower.

Reply:
But configuration it get's loaded before any command ever is created.

Reply:
But you can't (and should not inject) `IConfiguration` into an `ICommandArgument` cause you need arguments to load the configuration. Technically you could do that in `PretzelBaseCommandArguments.BindingCompleted`, but that feels wrong cause you would break the `Liskov substitution principle`.

Reply:
Ooooooooooh ok, I get it now, sorry!